### PR TITLE
Ensure pihole-FTL process is shut down cleanly on container exit / add tini

### PR DESCRIPTION
**Pull request by PromoFaux**
### **What does this PR aim to accomplish?:**

While playing around with adding `tini` into the container, I realised that it wasn't a lot of good because `start.sh` will always be the sole child process of  `tini` - and it can only have one child.

As pihole-FTL is spawned as a child of `start.sh`, we would need to add in a trap to `start.sh` anyway, to forward that onto `pihole-FTL`

~~Seems like we can just skip adding `tini` (saving a whopping 2MB in the process 😏) and just trap for exits and shut down `pihole-FTL` first.~~

Revision: Include tini for it's zombie process reaping abilities. If it turns out to be more than needed, we can remove it in a future step

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

**Comment by edgd1er** [comment]
At first, I thought tini was the solution then I read [tini + bash 4.x](https://ahmet.im/blog/minimal-init-process-for-containers/)
Should stop function be the place to handle all processes that were start in the background (logrotate if any, crond, or any other daemon that may come ) ?
what about [zombie child reaping](https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/), could it be something that needs to be handled ? FTL may not have child process, but cron or logrotate may have through the scripts they run ?

**Comment by PromoFaux** [comment]
I have some things to play about with later. The `tini + bash 4.x` link is one of them!

File: src/Dockerfile [hunk 1]
```diff
@@ -27,6 +27,7 @@ RUN apk add --no-cache \
     psmisc \
     shadow \
     sudo \
+    tini \
```
Comment (yubiuser) [inline src/Dockerfile:30]: Didn't you want to save 2 MB?...

**Review by yubiuser — COMMENTED** [review]
Why are you adding the `uninstall.sh` to the image?

If you want to same even the last bytes you could use a multi-stage build with one stage where you only download the repos to and the final stage where you only the copy the necessary data from the repo....

**Comment by PromoFaux** [comment]
> Why are you adding the `uninstall.sh` to the image?

It's an oversight - up until recently this has been chucking shit at the wall to see what sticks. There is still work to do in optimising the image, but it's being actively developed so, things will change.

The multistage build is something I'm toying with offline - (pretty sure I've mentioned this in one of the issues)

File: src/Dockerfile (hunk 1 above)
Comment (PromoFaux) [reply src/Dockerfile:30]: The smirking emoji was supposed to be representative of my not entire serious nature when saying this.

**Review by yubiuser — CHANGES_REQUESTED** [review]
Merge conflicts ahead

**Comment by PromoFaux** [comment]
Merging sans-approval

**Pull request merged**